### ISSUE TITLE
Int4 version not supported with latest huggingface GPTQ integration.

### ISSUE THREAD

**Issue by monksgoyal** [QwenLM/Qwen#243]
### 是否已有关于该错误的issue或讨论？ | Is there an existing issue / discussion for this?

- [X] 我已经搜索过已有的issues和讨论 | I have searched the existing issues / discussions

### 该问题是否在FAQ中有解答？ | Is there an existing answer for this in FAQ?

- [X] 我已经搜索过FAQ | I have searched FAQ

### 当前行为 | Current Behavior

model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen-7B-Chat-Int4", device_map="auto", trust_remote_code=True)
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen-7B-Chat-Int4", trust_remote_code=True)

giving error:
Qwen/Qwen-7B-Chat-Int4 does not appear to have a file named pytorch_model.bin, tf_model.h5, model.ckpt or flax_model.msgpack.

New Huggingface GPTQ integration source: https://huggingface.co/blog/gptq-integration

### 运行环境 | Environment

```Markdown
- OS:
- Python:
- Transformers:
- PyTorch:
- CUDA (`python -c 'import torch; print(torch.version.cuda)'`):
```

**Comment by JustinLin610** [comment]
Why not directly follow our guidance to directly use AutoGPTQ? Check the readme for the usage. This is equivalent to the very recently published new HF support of AutoGPTQ. I'll try it later but users have to change the version of HF to the latest.

**Comment by monksgoyal** [comment]
Yes that works without an issue but reason i want to use huggingface that we can do PEFT finetuning easily without changing code a lot.

**Comment by JustinLin610** [comment]
> Yes that works without an issue but reason i want to use huggingface that we can do PEFT finetuning easily without changing code a lot.

I'll give a try today to merge it with the latest Transformers

**Comment by JustinLin610** [comment]
> Yes, web demo would be fine!
> 
> I cloned the repository yesterday; I believe I have the latest code. I will check it again.
> 
> Thanks, George J.

CLI demo is also available for Int4 model now

**Comment by gcapnias** [comment]
Hi,

I tried the latest version of web_demo.py. I used the following command line - `python.exe .\web_demo.py --checkpoint-path 'Qwen/Qwen-7B-Chat-Int4' --inbrowser`.

I still get the same error running the script. 

On line 50, it seems the script does not find the file `quantize_config.json`, and it still tries to load the model from the else branch using **AutoModelForCausalLM**.

I tried to comment out the if statement (lines 50-60), leaving lines 51-59 to correctly load the model. I had to install manually `auto_gptq` library as it is not included in `requirements_web_demo.txt` with `pip`.

It still gives me an error: `RuntimeError: no device index` in loading the model, which I have not figure out what is the problem.


Regards,
George J.
